corehq/apps/app_manager/models.py

```diff
@@ -4133,8 +4133,11 @@ def post_url(self):
     def key_server_url(self):
         return reverse('key_server_url', args=[self.domain])
 
+    def heartbeat_url(self, build_profile_id=None):
+        return self.base_heartbeat_url + '?build_profile_id=%s' % (build_profile_id or '')
```

Comment:
this change isn't necessary since it will be added by mobile:
> For Heartbeat, Mobile will just include the buildProfileId from Profile as a query param in the request with key build_profile_id. (https://dimagi-dev.atlassian.net/browse/ICDS-909)

Reply:
oh, i remember @shubham1g5 mentioning that HQ would need to update this in the profile file. Will let him confirm.

Reply:
Simon is right that it will be added by mobile. HQ just provide the base url and mobile will append any params needed for HQ to process. This would have been required if we chose to not send the buildProfileId in `profile` node but since we are doing that heartbeat doesn't need to change on HQ.

Reply:
thanks @shubham1g5 and thanks @snopoke 
Okay.
Wont this result in delay of this information being available?
Considering this would be a mobile update, hence new LTS and hence a new version making it to the field?
If HQ does it would be available right off the next app update.

Reply:
That's a valid point, something that didn't cross my earlier and would also simplify the overall solution without the need of adding the `buildProfileId` in the `profile` node. I don't see any drawbacks as such of adding the build profile Id directly to Heartbeat url on HQ but tagging @ctsims  to pick his mind on it.

Reply:
We've followed the approach in the past of adding parameters to URL's like this directly, which is how app aware syncs work and a few other random pieces here and there. 

I think it's totally fine to add it here like this since 'profile' is effectively an HQ concept (rather than a mobile concept) in the first place. 

The mobile can use a URL builder to override the default behavior of the string if it , say, gets stale or otherwise needs to be changed on device to meet new behavior.

Reply:
@snopoke let me know if its now okay to leave it here.

Reply:
@ctsims  Makes sense. Since this would be supported out of the box on mobile, I don't think it makes sense for HQ to include `buildProfileId` in `profile` node anymore since the only place mobile was going to use it was for sending it in heartbeat.

Reply:
Fine with me

Reply:
thanks all.

[This is the PR](https://github.com/dimagi/commcare-hq/pull/25627) where we added the build profile id for mobile to use. Going to revert. 
We can add it back if mobile needs it for some other use case.